### PR TITLE
add jobs for new guest repos

### PR DESCRIPTION
**Pull request by hopkiw**
depends on: #726

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml [hunk 1]
```diff
@@ -37,7 +60,7 @@ postsubmits:
       workdir: true
     decorate: true
     branches:
-    - ^master$
+    - ^main$
```
Comment (hopkiw) [inline prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml:63]: new github repos use 'main'; we should migrate existing repos to match.

**Review by wrigri — APPROVED** [review]
/lgtm

**Review by wrigri — APPROVED** [review]
/lgtm

**Pull request merged**